### PR TITLE
[Impeller] Do not build scene unless 3d define is true

### PR DESCRIPTION
**Pull request by dnfield**
Fixes https://github.com/flutter/flutter/issues/133193

### CODE REVIEW

**Review by jonahwilliams — APPROVED** [review]
LGTM

File: impeller/aiks/aiks_unittests.cc [hunk 1]
```diff
@@ -2056,6 +2056,7 @@ TEST_P(AiksTest, SaveLayerFiltersScaleWithTransform) {
   ASSERT_TRUE(OpenPlaygroundHere(canvas.EndRecordingAsPicture()));
 }
 
+#if IMPELLER_ENABLE_3D
```
Comment (bdero) [inline impeller/aiks/aiks_unittests.cc:2059]: We should activate this flag in a CI build as part of this patch, otherwise Scene will just bitrot.

**Review by bdero — CHANGES_REQUESTED** [review]

File: impeller/aiks/aiks_unittests.cc (hunk 1 above)
Comment (dnfield) [reply impeller/aiks/aiks_unittests.cc:2059]: I've updated the mac unpt build (which AFAIU runs the impeller unittests) to add `--impeller-enable-3d`.

File: ci/builders/mac_unopt.json [hunk 2]
```diff
@@ -13,7 +13,8 @@
                 "debug",
                 "--unoptimized",
                 "--no-lto",
-                "--prebuilt-dart-sdk"
+                "--prebuilt-dart-sdk",
+                "--impeller-enable-3d"
```
Comment (bdero) [inline ci/builders/mac_unopt.json:17]: Will this unopt build get collected into artifacts that are shipped to Flutter users in any form? The `--impeller-enable-3d` flag also switches on the "experimental" version of `dart:ui`, which allows for appending scene nodes that we should continue not exposing to Flutter users.

Comment (jonahwilliams) [reply ci/builders/mac_unopt.json:17]: we don't ship unopt artifacts

**Review by bdero — APPROVED** [review]

File: ci/builders/mac_unopt.json (hunk 2 above)
Comment (zanderso) [inline ci/builders/mac_unopt.json:17]: The command to run tests for `host_debug_unopt` is defined down on line 167. Are there any additional tests to add there?

Comment (dnfield) [reply ci/builders/mac_unopt.json]: as long as impeller_unittests runs it's fine - which it should as part of run_tests.py on this platform

**Comment by dnfield** [comment]
I'ms eeing a vulkan validation failure on mac with this patch and don't know why - I'm trying to update to ToT and see if that fixes things.

**Comment by dnfield** [comment]
Coincidentally this is the same fraembuffer issue as we're seeing on the A02...

**Comment by dnfield** [comment]
https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8771905154412618609/+/u/test:_Host_Tests_for_host_debug/stdout#L9210_12

**Comment by dnfield** [comment]
Ok, I backed out an unintentional file.

**Comment by dnfield** [comment]
Here's a CI build that ran the Scene unit test:

https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8771859664658821969/+/u/test:_Host_Tests_for_host_debug_unopt/stdout#L9477_1

**Comment by chinmaygarde** [comment]
The web framework test failure seemed like a flake. Retrying.

**Pull request merged**